AndroidProblem/app/src/main/java/com/example/android/androidproblem/MainActivity.kt

```diff
@@ -0,0 +1,82 @@
+package com.example.android.androidproblem
+
+import android.content.SharedPreferences
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import android.widget.Button
+import android.widget.EditText
+import android.widget.ListView
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+
+class MainActivity : AppCompatActivity() {
+    val emailId by lazy { findViewById<EditText>(R.id.email_id) }
+    val userList by lazy { findViewById<ListView>(R.id.user_list) }
+    val done by lazy { findViewById<Button>(R.id.done_button) }
+    val userString = ArrayList<String>()
+    val listOfImageUrl = ArrayList<String>()
+    val apiService by lazy { RestApiService() }
+    val gson = Gson()
+    lateinit var sharedPref:SharedPreferences
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        GlobalScope.launch{
+            getCache()
+        }
+        done.setOnClickListener {
+            if (emailId.text.isNullOrEmpty()) {
+                Log.d(TAG, "invalid email")
```

Comment:
This log is not visible to user? We need to tell user that you need to type something in email field.
May be use a toast?